gum/backend-qnx/gumtls-qnx.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2015 Ole André Vadla Ravnås <oleavr@nowsecure.com>
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+
+#include "gumprocess.h"
+#include "gumspinlock.h"
+#include "gumtls.h"
+
+#include <pthread.h>
+#include <sys/syspage.h>
+
+#define MAX_TMP_TLS_KEYS 200
+typedef struct _GumTmpTlsKey GumTmpTlsKey;
+
+struct _GumTmpTlsKey{
+  GumThreadId tid;
+  GumTlsKey key;
+  gpointer value;
+};
+
+static GumTmpTlsKey _gum_tls_tmp_keys[MAX_TMP_TLS_KEYS];
+static GumSpinlock _gum_tls_tmp_keys_lock;
+
+static gpointer _gum_tls_key_get_tmp_value (GumTlsKey key);
+static void _gum_tls_key_set_tmp_value (GumTlsKey key, gpointer value);
+static void _gum_tls_key_delete_tmp_value (GumTlsKey key);
+
+void
+_gum_tls_init (void)
+{
+  gum_spinlock_init (&_gum_tls_tmp_keys_lock);
+  memset (_gum_tls_tmp_keys, 0, sizeof (_gum_tls_tmp_keys));
+}
+
+static gpointer
```

Comment:
The *tmp_value* functions are lower level and should go to the bottom of the file, so the public API (high-level stuff) is at the top.